core/templates/dev/head/pages/exploration_player/PlayerServices.js

```diff
@@ -271,7 +271,11 @@ oppia.factory('oppiaPlayerService', [
               classificationResult.ruleSpecIndex);
           }
 
-          var outcome = classificationResult.outcome;
+          // Use Object.assign to clone the object
+          // since classificationResult.outcome points
+          // at oldState.interaction.default_outcome
+          var outcome = Object.assign({}, classificationResult.outcome);
```

Comment:
I agree, angular.copy() is better, didn't know about it before. PR updated with commit.

And thanks for the kind words!